Client/Helpers/MsalAuthenticationProvider.cs

```diff
@@ -12,11 +14,11 @@ namespace DeltaQueryClient
     // the GraphSDK team.  It will supports all the types of Client Application as defined by MSAL.
     public class MsalAuthenticationProvider : IAuthenticationProvider
     {
-        private PublicClientApplication _clientApplication;
-        private string[] _scopes;
-        private string _accessToken = null;
+        private IPublicClientApplication _clientApplication;
+        private IEnumerable<string> _scopes;
```

Comment:
this class is not very reusable as the developer passes scopes upfront.
Another way of doing would be to set the scopes on the client before calling an API ...